pkg/ingester/ingester.go

```diff
@@ -890,23 +894,35 @@ func (i *Ingester) PushWithCleanup(ctx context.Context, pushReq *push.Request) (
 	i.appendedSamplesStats.Inc(int64(succeededSamplesCount))
 	i.appendedExemplarsStats.Inc(int64(succeededExemplarsCount))
 
+	group := ""
+	if len(req.Timeseries) > 0 {
+		group = validation.FindGroupLabel(i.limits, userID, req.Timeseries[0].Labels)
```

Comment:
The `len(req.Timeseries) > 0` condition is repeatead everywhere. Can we move that into the group extraction function and make code cleaner everywhere?